pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -168,9 +169,12 @@ func (c *Cluster) Start() error {
 	errs := newProvisionErrors()
 
 	// Validate pod's memory if specified
-	err := controller.CheckPodMemory(cephv1.ResourcesKeyOSD, cephv1.GetOSDResources(c.spec.Resources), cephOsdPodMinimumMemory)
-	if err != nil {
-		return errors.Wrap(err, "failed to check pod memory")
+	osdResourceKeyOSDs := []string{cephv1.ResourcesKeyOSD, cephv1.ResourcesKeyHDDDeviceOSD, cephv1.ResourcesKeySSDDeviceOSD, cephv1.ResourcesKeyNVMEDeviceOSD}
+	for _, osdKey := range osdResourceKeyOSDs {
```

Comment:
Instead of iterating over these specific 4 keys, let's iterate over all the keys that have the prefix `osd`. Otherwise, we will miss validation of custom device classes.